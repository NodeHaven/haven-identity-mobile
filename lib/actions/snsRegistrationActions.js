// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  REGISTER_DEVICE_FOR_NOTIFICATIONS,
  RECEIVE_DEVICE_TOKEN,
  STORE_ENDPOINT_ARN,
  STORE_DEVICE_TOKEN,
  SKIP_PUSH_NOTIFICATIONS,
  SEND_LOCAL_NOTIFICATION,
  CANCEL_SCHEDULED_NOTIFICATION,
  UPDATE_BADGE_COUNT,
  HANDLE_ENCRYPTED_MESSAGE,
  CHECK_FOR_NOTIFICATIONS,
  POLL_FOR_NOTIFICATIONS,
  INIT_NOTIFICATIONS
} from 'uPortMobile/lib/constants/SnsRegistrationActionTypes'

export function registerDeviceForNotifications () {
  return {
    type: REGISTER_DEVICE_FOR_NOTIFICATIONS
  }
}

export function skipPushNotifications () {
  return {
    type: SKIP_PUSH_NOTIFICATIONS
  }
}

export function storeEndpointArn (endpointArn) {
  return {
    type: STORE_ENDPOINT_ARN,
    endpointArn
  }
}

export function receiveDeviceToken (deviceToken) {
  return {
    type: RECEIVE_DEVICE_TOKEN,
    deviceToken
  }
}

export function storeDeviceToken (deviceToken) {
  return {
    type: STORE_DEVICE_TOKEN,
    deviceToken
  }
}

export function sendLocalNotification (activityId, message, repeatInterval) {
  return {
    type: SEND_LOCAL_NOTIFICATION,
    message,
    activityId,
    repeatInterval
  }
}

export function cancelScheduledNotification (activityId) {
  return {
    type: CANCEL_SCHEDULED_NOTIFICATION,
    activityId
  }
}

export function updateBadgeCount () {
  return {
    type: UPDATE_BADGE_COUNT
  }
}

export function checkForNotifications () {
  return {
    type: CHECK_FOR_NOTIFICATIONS
  }
}

export function pollForNotifications () {
  return {
    type: POLL_FOR_NOTIFICATIONS
  }
}

export function handleEncryptedMessage (id) {
  return {
    type: HANDLE_ENCRYPTED_MESSAGE,
    id
  }
}

export function initNotifications () {
  return {
    type: INIT_NOTIFICATIONS
  }
}
