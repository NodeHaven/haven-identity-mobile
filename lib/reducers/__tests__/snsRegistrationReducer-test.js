// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//

import reducer from 'uPortMobile/lib/reducers/snsRegistrationReducer.js'
import {storeEndpointArn, storeDeviceToken, skipPushNotifications} from 'uPortMobile/lib/actions/snsRegistrationActions.js'

const empty = {}
const withEndpointArn = {endpointArn: 'aws:arn:b'}
const withDeviceToken = {deviceToken: 'yyyyyy'}
const skippedPushNotifications = {skippedPushNotifications: true}

it('sets endpoint arn', () => {
  expect(reducer(null, storeEndpointArn('aws:arn:a'))).toMatchSnapshot()
  expect(reducer(empty, storeEndpointArn('aws:arn:a'))).toMatchSnapshot()
  expect(reducer(withEndpointArn, storeEndpointArn('aws:arn:a'))).toMatchSnapshot()
  expect(reducer(withDeviceToken, storeEndpointArn('aws:arn:a'))).toMatchSnapshot()
  expect(reducer(null, storeEndpointArn('aws:arn:a'))).toMatchSnapshot()
})

it('sets device token', () => {
  expect(reducer(null, storeDeviceToken('xxxxxx'))).toMatchSnapshot()
  expect(reducer(empty, storeDeviceToken('xxxxxx'))).toMatchSnapshot()
  expect(reducer(withEndpointArn, storeDeviceToken('xxxxxx'))).toMatchSnapshot()
  expect(reducer(withDeviceToken, storeDeviceToken('xxxxxx'))).toMatchSnapshot()
  expect(reducer(null, storeDeviceToken('xxxxxx'))).toMatchSnapshot()
})

it('sets skipPushNotifications', () => {
  expect(reducer(null, skipPushNotifications())).toMatchSnapshot()
  expect(reducer(empty, skipPushNotifications())).toMatchSnapshot()
  expect(reducer(withEndpointArn, skipPushNotifications())).toMatchSnapshot()
  expect(reducer(withDeviceToken, skipPushNotifications())).toMatchSnapshot()
  expect(reducer(null, skipPushNotifications())).toMatchSnapshot()
})

it('ignores unsupported action', () => {
  expect(reducer(undefined, {type: 'UNSUPPORTED'})).toEqual(empty)
  expect(reducer(empty, {type: 'UNSUPPORTED'})).toEqual(empty)
  expect(reducer(withEndpointArn, {type: 'UNSUPPORTED'})).toEqual(withEndpointArn)
  expect(reducer(withDeviceToken, {type: 'UNSUPPORTED'})).toEqual(withDeviceToken)
  expect(reducer(skippedPushNotifications, {type: 'UNSUPPORTED'})).toEqual(skippedPushNotifications)
})
