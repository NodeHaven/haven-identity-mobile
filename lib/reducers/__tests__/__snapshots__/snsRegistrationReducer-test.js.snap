// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets device token 1`] = `
Object {
  "deviceToken": "xxxxxx",
}
`;

exports[`sets device token 2`] = `
Object {
  "deviceToken": "xxxxxx",
}
`;

exports[`sets device token 3`] = `
Object {
  "deviceToken": "xxxxxx",
  "endpointArn": "aws:arn:b",
}
`;

exports[`sets device token 4`] = `
Object {
  "deviceToken": "xxxxxx",
}
`;

exports[`sets device token 5`] = `
Object {
  "deviceToken": "xxxxxx",
}
`;

exports[`sets endpoint arn 1`] = `
Object {
  "endpointArn": "aws:arn:a",
}
`;

exports[`sets endpoint arn 2`] = `
Object {
  "endpointArn": "aws:arn:a",
}
`;

exports[`sets endpoint arn 3`] = `
Object {
  "endpointArn": "aws:arn:a",
}
`;

exports[`sets endpoint arn 4`] = `
Object {
  "deviceToken": "yyyyyy",
  "endpointArn": "aws:arn:a",
}
`;

exports[`sets endpoint arn 5`] = `
Object {
  "endpointArn": "aws:arn:a",
}
`;

exports[`sets skipPushNotifications 1`] = `
Object {
  "skippedPushNotifications": true,
}
`;

exports[`sets skipPushNotifications 2`] = `
Object {
  "skippedPushNotifications": true,
}
`;

exports[`sets skipPushNotifications 3`] = `
Object {
  "endpointArn": "aws:arn:b",
  "skippedPushNotifications": true,
}
`;

exports[`sets skipPushNotifications 4`] = `
Object {
  "deviceToken": "yyyyyy",
  "skippedPushNotifications": true,
}
`;

exports[`sets skipPushNotifications 5`] = `
Object {
  "skippedPushNotifications": true,
}
`;
